Source/WebCore/style/CustomPropertyRegistry.cpp

```diff
@@ -194,5 +188,50 @@ void CustomPropertyRegistry::notifyAnimationsOfCustomPropertyRegistration(const
     }
 }
 
+auto CustomPropertyRegistry::parseInitialValue(const Document& document, const AtomString& propertyName, const CSSCustomPropertySyntax& syntax, CSSParserTokenRange tokenRange) -> Expected<std::pair<RefPtr<CSSCustomPropertyValue>, ViewportUnitDependency>, ParseInitialValueError>
+{
+    // FIXME: This parses twice.
+    auto dependencies = CSSPropertyParser::collectParsedCustomPropertyValueDependencies(syntax, tokenRange, strictCSSParserContext());
+    if (!dependencies.isComputationallyIndependent())
+        return makeUnexpected(ParseInitialValueError::NotComputationallyIndependent);
+
+    // We don't need to provide a real context style since only computationally independent values are allowed (no 'em' etc).
+    auto placeholderStyle = RenderStyle::create();
+    Style::Builder builder { placeholderStyle, { document, RenderStyle::defaultStyle() }, { }, { } };
+
+    auto initialValue = CSSPropertyParser::parseTypedCustomPropertyInitialValue(propertyName, syntax, tokenRange, builder.state(), { document });
+    if (!initialValue)
+        return makeUnexpected(ParseInitialValueError::DidNotParse);
+
+    return std::pair { WTFMove(initialValue), dependencies.viewportDimensions ? ViewportUnitDependency::Yes : ViewportUnitDependency::No };
+}
+
+void CustomPropertyRegistry::invalidatePropertiesWithViewportUnits(Document& document)
+{
+    bool invalidatedAny = false;
+
+    auto invalidatePropertiesWithViewportUnits = [&](auto& map) {
+        for (auto& property : map.values()) {
```

Comment:
We could check (or assert) for `property` before using it.

Reply:
We never add null to the maps. That calls for making the type `UniqueRef` instead of `unique_ptr`.